### ISSUE TITLE
Proposal: Include Thing in domain for ListItem/nextItem

### ISSUE THREAD

**Issue by jaygray0919** [schemaorg/schemaorg#1170]
The expected type for ListItem/item is Thing.
So, for example, this is valid:

```
<script type="application/ld+json">
{
"@context":
    {
    "@vocab": "http://schema.org/",
    "id": "@id",
    "graph": "@graph",
    "type": "@type"
    },
"graph":
    [
        {
        "type": "ItemList",
        "id": "Intangible/ItemList/1001",
        "name": "Chemicals",
        "description": "Chemicals used in example",
        "itemListElement":
        [
            {
            "type": "ListItem",
            "position": 1,
            "item":
                {
                "type": "Thing",
                "id": "http://lod.nal.usda.gov/nalt/214505"
                }
            },
            {
            "type": "ListItem",
            "position": 2,
            "item":
                {
                "type": "Thing",
                "id": "http://lod.nal.usda.gov/nalt/7862"
                }
            }
        ]
        },
        {
        "type": "Thing",
        "id": "http://lod.nal.usda.gov/nalt/214505",
        "name": "cytidine deaminase"
        },
        {
        "type": "Thing",
        "id": "http://lod.nal.usda.gov/nalt/7862",
        "name": "aminoethoxyvinylglycine"
        }
    ]
}
</script>
```

Now let's add nextItem. `nextItem` expects the Type `ListItem`.
But if `nextItem` also is defined as `Thing`, the model breaks. GSDTT throws an error about nexItem is not a valid target type for nextItem.

Here is the non-valid example:

```
<script type="application/ld+json">
{
"@context":
    {
    "@vocab": "http://schema.org/",
    "id": "@id",
    "graph": "@graph",
    "type": "@type"
    },
"graph":
    [
        {
        "type": "ItemList",
        "id": "Intangible/ItemList/1001",
        "name": "Chemicals",
        "description": "Chemicals used in example",
        "itemListElement":
        [
            {
            "type": "ListItem",
            "position": 1,
            "item":
                {
                "type": "Thing",
                "id": "http://lod.nal.usda.gov/nalt/214505"
                },
            "nextItem":
                {
                "type": "ListItem",
                "id": "http://lod.nal.usda.gov/nalt/7862"
                }
            },
            {
            "type": "ListItem",
            "position": 2,
            "item":
                {
                "type": "Thing",
                "id": "http://lod.nal.usda.gov/nalt/7862"
                }
            }
        ]
        },
        {
        "type": "Thing",
        "id": "http://lod.nal.usda.gov/nalt/214505",
        "name": "cytidine deaminase"
        },
        {
        "type": "Thing",
        "id": "http://lod.nal.usda.gov/nalt/7862",
        "name": "aminoethoxyvinylglycine"
        }
    ]
}
</script>

```

It seems to us that the solution is to allow `nextItem` to be a `Thing` since the LOGICAL next item is an `item` and may be a `Thing`.

Are we missing something here, or doing something wrong?

/jay gray

**Comment by jvandriel** [comment]
What's your motivation for wanting to expand the range of ```nextItem```/```previousItem``` to ```Thing```? Is there something particular you're trying to resolve?

**Comment by jaygray0919** [comment]
Thank you for asking. In the above example, we would like to create an `ordered list`. Items in an ordered list are `Thing`s.

We would like the ordered list to be structured, and not dependent on using a literal value for the key 'position'. So our thinking was: use `nextItem` to specify the ordering. In some ways, `nextItem` is the inverse of `supercededBy` for `@Class` or `@Property`.

At a higher level, we need to specify a 'parent', a 'sibling' and a 'child' relationship - in effect, the ordering of generic  `hasPart` and `isPartOf` relationships.

Ping me if we need to demonstrate more specific examples.

**Comment by jvandriel** [comment]
Just to make sure I understand your use case correctly, would this markup get you what you're looking for?

```
<script type="application/ld+json">
{
  "@context":
  {
    "@vocab":"http://schema.org",
    "id":"@id",
    "graph":"@graph",
    "type":"@type"
  },
  "graph":
  [
    {
	  "type": "ItemList",
      "id": "Intangible/ItemList/1001",
      "name": "Chemicals",
      "description": "Chemicals used in example",
      "itemListElement":
      [
        {
          "type":"ListItem",
	      "id":"#ListItem-1",
          "item":{"id":"http://lod.nal.usda.gov/nalt/214505"},
          "nextItem":{"id":"#ListItem-2"}
        },
        {
          "type":"ListItem",
          "id":"#ListItem-2",
	      "item":{"id":"http://lod.nal.usda.gov/nalt/7862"},
	      "previousItem":{"id":"#ListItem-1"}
        }
      ]
	},
	{
      "type":"Thing",
      "id":"http://lod.nal.usda.gov/nalt/214505",
      "name": "cytidine deaminase"
    },
    {
      "type": "Thing",
      "id": "http://lod.nal.usda.gov/nalt/7862",
      "name": "aminoethoxyvinylglycine"
    }
  ]
}
</script>
```

**Comment by RichardWallis** [comment]
Or perhaps a [visually] simpler alternative using MTEs:
```
<script type="application/ld+json">
{
	"@context": {
		"@vocab": "http://schema.org",
		"id": "@id",
		"graph": "@graph",
		"type": "@type"
	},
	"graph": [
		{
			"type": "ItemList",
			"id": "Intangible/ItemList/1001",
			"name": "Chemicals",
			"description": "Chemicals used in example",
			"itemListElement": [
				{
					"type": "Thing ListItem",
					"id": "http://lod.nal.usda.gov/nalt/214505",
					"name": "cytidine deaminase",
					"nextItem": {
						"id": "http://lod.nal.usda.gov/nalt/7862"
					}
				},
				{
					"type": "Thing ListItem",
					"id": "http://lod.nal.usda.gov/nalt/7862",
					"name": "aminoethoxyvinylglycine",
					"previousItem": {
						"id": "http://lod.nal.usda.gov/nalt/214505"
					}
				}
			]
		}
	]
}
</script>
```

**Comment by jvandriel** [comment]
mmm, I don't really feel comfortable with your example @RichardWallis.

First off it bypasses the ```ListItem```s ```item``` property plus I'm not convinced it's wise to start expressing a thing is an ```ListItem``` as well as something else. Seems a bit far fetched to me.

**Comment by RichardWallis** [comment]
@jvandriel Can see where your uncomfort is coming from.

However in this world where MTE's are now more acceptable, ```item``` feels like an unnecessary  indirection to me.

**Comment by jvandriel** [comment]
I get where you're coming from Richard, and I'm the last person that's opposed to using MTEs (especially since I use them quite happily). What I'm wondering though is whether in this case the meaning of things stays correct. Specifying a book is a ["Product","Book"] feels non-conflicting to me, where as ["ListItem","Book"] feels 'weird'.

Than again, maybe that's just me.

**Comment by thadguidry** [comment]
@jvandriel Its just you. :) I can have a book item in my list of things.  I can have any Thing in my list of Things. And we already allow that and have that understanding at an "item" level.  What the consumer will understand from what the publisher listed will heavily depend on appropriate Types and Properties, as always.

I understand @jaygray0919 needs actually, which is around When and Where to properly use nextItem.  He is actually using it correctly.  Its just that GSDTT is not aware that doing extra service and providing the explicit type of Thing on "item" is perfectly valid.  @jaygray0919 should be allowed to be explicit and provide extra Types to further describe their "item"-like Thing including mentioning that it is a Type of Thing if it makes it easier to read or bring meaning to his structure or a consuming client.

In summary : GSDTT just needs a quick fix and not throw an error if a publisher happens to be explicit with their Typing when a Type is already expected, like that of "item" already expecting a value of Thing
https://schema.org/item
```
            "item":
                {
                "type": "Thing",
                "id": "http://lod.nal.usda.gov/nalt/7862"
                }
```

**Comment by jaygray0919** [comment]
TY @thadguidry - wiIl make our life easier. We have a comprehensive specification of `Thing`s and want to be able to reuse their specification in an `ItemList`. In our case the `Thing`s are identified by an `IRI`, and we want to re-use our `IRI`s in different scenarios.
We face the problem developers frequently present on this forum - we think our semantic design is correct but it fails GSDTT validation. @danbri chafes at that approach, makes a compelling case, and encourages use of JSON-LD Playground as the validator. But, ATEOTD, our purpose is to present content that can be harvested by tools like GSDTT. So if it fails GSDTT, it's wrong. So thanks for the correction.
/jay

**Comment by jvandriel** [comment]
> "Its just you. :)"

See, that does it for me already - great way of stopping me from dragging on a discussion when it isn't needed (simply because I'm the one that isn't getting something).    :)

So just to confirm if I get you all correctly - hehe - if there's no issue with skipping `the ``item``` property than the following would be correct, right?

```
<script type="application/ld+json">
{
  "@context":
  {
    "@vocab":"http://schema.org",
    "id":"@id",
    "graph":"@graph",
    "type":"@type"
  },
  "graph":
  [
    {
      "type":"ItemList",
      "id":"Intangible/ItemList/1001",
      "name":"Chemicals",
      "description":"Chemicals used in example",
      "itemListElement":
      [
        {
          "type":"ListItem",
          "id":"http://lod.nal.usda.gov/nalt/214505",
          "nextItem":{"id":"http://lod.nal.usda.gov/nalt/7862"}
        },
        {
          "type":"ListItem",
          "id":"http://lod.nal.usda.gov/nalt/7862",
          "previousItem":{"id":"http://lod.nal.usda.gov/nalt/214505"}
        }
      ]
    },
    {
      "type":"Thing",
      "id":"http://lod.nal.usda.gov/nalt/214505",
      "name":"cytidine deaminase"
    },
    {
      "type":"Thing",
      "id":"http://lod.nal.usda.gov/nalt/7862",
      "name":"aminoethoxyvinylglycine"
    }
  ]
}
</script>
```  

Does this also come close enough to what you're hoping to do @jaygray0919 ?

_notre: The new and improved Google Structured Data Testing Tool already handles it fine, with the exception of some errors that are about Google specific features - and the Structured Data Linter doesn't report any issues at all._

**Comment by jvandriel** [comment]
_"we think our semantic design is correct but it fails GSDTT validation."_

Well, I can't say I haven't spend my fair share of time on pulling out hairs while uttering words I won't repeat here  - nevertheless it has to be said that over the last year Google's Structured Data Testing Tool has made huge steps forward in regards to what theoretically can be done and what the SDTT is able to handle. 

IMO the folks responsible for this should be applauded for once as well (we all know they already got more than their share of negative feedback) - So at least from me: **Thank you, you [anonymous] group of developers that work on the SDTT. You did us all a big favor - especially now the SDTT also handles MTEs correctly!**

**Comment by jaygray0919** [comment]
I certainly did not mean to throw shade on GSDTT or its developers - and I'm in complete agreement with your above comments. However, being an advanced schema.org developer and a frequent GSDTT 'validator', we have encountered many errors in the past. When we encounter those errors, we're not sure "who's at fault." Since schema.org is 'open source' and GSDDT is a Google product, @danbri faithfully separates 'church' and 'state' on this forum. But as developers, we often can't determine if we have a 'false negative' or a 'real negative'. There is a very responsible commentator on the Google Webmaster site (Grace Massa Langlois) who helps separate false-negatives from real-negatives, but we still encounter uncertainty - as with the source of this thread.

While I'm in this mood, let me also say that @RichardWallis has directly/indirectly helped us many times with commentary and advice.

Nevertheless, for us, GSDTT is and must be the ultimate test. So we need to be able to present test cases here, and ask/hope that GSDTT-experts will fix errors, or help correct our errors.

WRT @jvandriel revised code - the answer is yes, that is what we're hoping to do. Now let me take one more step with the example, Imagine a list that is an outline:
  1
  1.1
  1.2
  1.2.1
  1.2.2

In this case, the `nextItem` for 1.2 is an array. The `previousItem` for both 1.2.1 and 1.2.2 is 1.2
In the @jvandriel example, `previousItem` does the same work as `supersededBy` for `@Class` and `@Property`. IOHO, this is goodness; terms can do the same work (using different labels) for different `@Types`.

However, there is no `nextItem` equivalent for `@Class` and `@Property`.
IOHO, there should be.

In both cases, we want to relate a 'semantic item' (a schema.org `@Thing` with appropriate mark-up and an IRI identifier) with the `ItemList` or the `@Class`/`@Property`.

**Comment by gkellogg** [comment]
@jvandriel Your example is correct, but because both the ListItem entries and the `Thing` definitions use the same identifier, it's semantically indistinguishable from the following:

```json
<script type="application/ld+json">
{
  "@context":
  {
    "@vocab":"http://schema.org",
    "id":"@id",
    "graph":"@graph",
    "type":"@type"
  },
    "type":"ItemList",
    "id":"Intangible/ItemList/1001",
    "name":"Chemicals",
    "description":"Chemicals used in example",
    "itemListElement":
    [
      {
        "type":["ListItem", "Thing"],
        "id":"http://lod.nal.usda.gov/nalt/214505",
        "nextItem":{"id":"http://lod.nal.usda.gov/nalt/7862"},
        "name":"cytidine deaminase"
      },
      {
        "type":["ListItem", "Thing"],
        "id":"http://lod.nal.usda.gov/nalt/7862",
        "previousItem":{"id":"http://lod.nal.usda.gov/nalt/214505"},
        "name":"aminoethoxyvinylglycine"
      }
    ]
  }
}
</script>
```

This will be lost when turning into RDF, or when re-framing the JSON-LD document in any case.

(Arguably, the `Thing` types can be eliminated too, as `ListItem` is a subclass of `Thing`, but this won't always be true). (Also, the `ListItem` type could be left out, as it can be inferred from the use of either `nextItem` or `previousItem`.

Also, as an aside, it makes absolutely no sense to me that `nextItem` and `previousItem` are not defined as `@type: @id`, as there is no case in which a literal value could ever be used here, which would reduce the markup overhead further.

**Comment by jvandriel** [comment]
I more or less get you @gkellogg, until about this far (which the Linter accepts but of course which Google's SDTT doesn't):

```
<script type="application/ld+json">
{
  "@context":
  {
    "@vocab":"http://schema.org/",
    "id":"@id",
    "graph":"@graph",
    "type":"@type"
  },
  "type":"id",
  "id":"Intangible/ItemList/1001",
  "name":"Chemicals",
  "description":"Chemicals used in example",
  "itemListElement":
  [
    {
      "type":"id",
      "id":"http://lod.nal.usda.gov/nalt/214505",
      "nextItem":{"id":"http://lod.nal.usda.gov/nalt/7862"},
      "name":"cytidine deaminase"
    },
    {
      "type":"id",
      "id":"http://lod.nal.usda.gov/nalt/7862",
      "previousItem":{"id":"http://lod.nal.usda.gov/nalt/214505"},
      "name":"aminoethoxyvinylglycine"
    }
  ]
}
</script>
```

Though I don't understand how I have to express the following in the markup. Mind giving the example?:

> Also,as an aside, it makes absolutely no sense to me that nextItem and previousItem are not defined as @type: @id
